### DISCUSSION TITLE
Failed to connect

### DISCUSSION THREAD

**Discussion by hormic77** [suchmememanyskill/CYD-Klipper#114] [Q&A]
Hello,

I installed the newest version on my 2,4" ESP32.
Initial WLAN configuration works great.
But I can't connect zu klipper. Error message is "Failed to connect".
I tried the moonraker port 7125 and also the default web port 80. Where Mainsail ist reachable without any issue (on other devices)

Lokal IP subnet ist wihtelisted in the moonraker config.

Have I missed something?

Thanks a lot!

**Comment by suchmememanyskill** [comment]
Try going to http://{moonraker_ip}:{moonraker_port}/server/info in a webbrowser on a device on the same network as cyd-klipper is on. Does this work?

**Reply by hormic77** [reply]
yes that works. Result is:
{
  "result": {
    "klippy_connected": true,
    "klippy_state": "ready",
    "components": [
      "secrets",
      "template",
      "klippy_connection",
      "jsonrpc",
      "internal_transport",
      "application",
      "websockets",
      "dbus_manager",
      "database",
      "file_manager",
      "authorization",
      "klippy_apis",
      "shell_command",
      "machine",
      "data_store",
      "proc_stats",
      "job_state",
      "job_queue",
      "history",
      "http_client",
      "announcements",
      "webcam",
      "extensions",
      "octoprint_compat",
      "update_manager"
    ],
    "failed_components": [],
    "registered_directories": [
      "config",
      "logs",
      "gcodes",
      "config_examples",
      "docs"
    ],
    "warnings": [],
    "websocket_count": 2,
    "moonraker_version": "v0.8.0-379-g346a3d7",
    "missing_klippy_requirements": [],
    "api_version": [1, 4, 0],
    "api_version_string": "1.4.0"
  }
}

**Reply by hormic77** [reply]
URL is: http://192.168.0.97:7125/server/info
Same network, same SSID

**Reply by suchmememanyskill** [reply]
And if you type '192.168.0.97' in the first field, '7125' in the second, it just refuses to connect?

**Reply by hormic77** [reply]
yes correct
Also reflased it several times.

**Reply by suchmememanyskill** [reply]
If you capture the logs (the web installer can do this), do you get any log message when you try to connect?

Do you get any log messages on the moonraker side? (If you disable the ip range protection, does it change anything?)

**Comment by hormic77** [comment]
today I looked deeper into everything.

The moonraker log did not show any sign of a connect or also not If I try to block the device via the config.
Klipper could ping the ESP Display. 
tcpdump did not show any packet to moonraker

That I created a hotspot on my mobile thats also connected to the local network.
Then ESP-Display came up with the API key question. So I whitelisted the ESP and after that it was successfull connected to klipper.

Then I disabled the hotspot and reconfigured wifi back up my normal network and ist was connected successfully again.

very strange!

But thanks a lot for your help! Now I need some time to play around with it!
